packages/wrapped-keys/update-ipfs-cids.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs');
+const path = require('path');
+
+const Hash = require('ipfs-only-hash');
+
+const {
+  code: batchGenerateEncryptedKey,
+} = require('../wrapped-keys-lit-actions/src/generated/common/batchGenerateEncryptedKeys');
+const {
+  code: exportPrivateKey,
+} = require('../wrapped-keys-lit-actions/src/generated/common/exportPrivateKey');
+const {
+  code: generateEncryptedEthereumPrivateKey,
+} = require('../wrapped-keys-lit-actions/src/generated/ethereum/generateEncryptedEthereumPrivateKey');
+const {
+  code: signMessageWithEncryptedEthereumKey,
+} = require('../wrapped-keys-lit-actions/src/generated/ethereum/signMessageWithEncryptedEthereumKey');
+const {
+  code: signTransactionWithEncryptedEthereumKey,
+} = require('../wrapped-keys-lit-actions/src/generated/ethereum/signTransactionWithEncryptedEthereumKey');
+const {
+  code: generateEncryptedSolanaPrivateKey,
+} = require('../wrapped-keys-lit-actions/src/generated/solana/generateEncryptedSolanaPrivateKey');
+const {
+  code: signMessageWithEncryptedSolanaKey,
+} = require('../wrapped-keys-lit-actions/src/generated/solana/signMessageWithEncryptedSolanaKey');
+const {
+  code: signTransactionWithEncryptedSolanaKey,
+} = require('../wrapped-keys-lit-actions/src/generated/solana/signTransactionWithEncryptedSolanaKey');
+
+async function updateConstants() {
+  const constantsPath = path.join(
+    __dirname,
+    './src/lib/lit-actions-client/constants.ts'
+  );
+  let constantsContent = fs.readFileSync(constantsPath, 'utf-8');
+
+  const newCIDs = {
+    signTransaction: {
+      evm: await Hash.of(signTransactionWithEncryptedEthereumKey),
+      solana: await Hash.of(signTransactionWithEncryptedSolanaKey),
+    },
+    signMessage: {
+      evm: await Hash.of(signMessageWithEncryptedEthereumKey),
+      solana: await Hash.of(signMessageWithEncryptedSolanaKey),
+    },
+    generateEncryptedKey: {
+      evm: await Hash.of(generateEncryptedEthereumPrivateKey),
+      solana: await Hash.of(generateEncryptedSolanaPrivateKey),
+    },
+    exportPrivateKey: {
+      evm: await Hash.of(exportPrivateKey),
+      solana: await Hash.of(exportPrivateKey),
+    },
+    batchGenerateEncryptedKeys: await Hash.of(batchGenerateEncryptedKey),
+  };
+
+  Object.entries(newCIDs).forEach(([key, value]) => {
```

Comment:
Dealing with injecting types and Object.freeze() here makes this code quite awkward and brittle.

Rather than doing string replacement operations, what do you think about us extracting the content of the `LIT_ACTION_CID_REPOSITORY` to a dedicated `.json` file that we can _literally replace the entire contents of_, from this script?

We would then import that JSON file into the `constants.ts` file in the lit-actions-client, _recursively call Object.freeze() on the object_, and export it with the LitCidRepository type annotation?  Incidentally, in order to export the object `as const` (which I'd like for us to do in the future) would require extracting the `Object.freeze()` calls out of the object definition also, because you can't mark an object that is wrapped in `Object.freeze()` with the `as const` pragma.

Then this entire constant object could be a _literal_ JSON object that is imported, typed, frozen, and exported for consumers by code inside the wrapped-keys constants file.

Reply:
Tried everything in a single operation on a TS file as `const`, I prefer JS/TS files over json. But it turned out with the same level of complexity so json it is. Updated